internal/models/post.go

```diff
@@ -6,24 +6,28 @@ import (
 )
 
 type Post struct {
-	Id        uuid.UUID `gorm:"column:id;primary_key"`
-	Username  string    `gorm:"column:username"`
-	User      User      `gorm:"foreignKey:username;references:username"`
-	Content   string    `gorm:"column:content;type:varchar(256);null"`
-	ImageUrl  string    `gorm:"column:image_url;type:varchar(256);null"`
-	Hashtags  []Hashtag `gorm:"many2many:post_hashtags;"` // gorm handles the join table
-	CreatedAt time.Time `gorm:"column:created_at;not_null"`
+	Id         uuid.UUID `gorm:"column:id;primary_key"`
+	Username   string    `gorm:"column:username"`
+	User       User      `gorm:"foreignKey:username;references:username"`
+	Content    string    `gorm:"column:content;type:varchar(256);null"`
+	ImageUrl   string    `gorm:"column:image_url;type:varchar(256);null"`
+	Hashtags   []Hashtag `gorm:"many2many:post_hashtags;"` // gorm handles the join table
+	CreatedAt  time.Time `gorm:"column:created_at;not_null"`
+	LocationId uuid.UUID `gorm:"column:locationId"`
+	Location   Location  `gorm:"foreignKey:locationId;references:locationId"`
 }
 
 type PostCreateRequestDTO struct {
-	Content string `json:"content" binding:"required"`
+	Content  string    `json:"content" binding:"required"`
+	Location *Location `json:"location" `
```

Comment:
Beim Create Request wird nicht die Location Entity übergeben (die erstellst du danach ja erst). Sondern ein LocationRequestDTO. Da LocationRequestDTO und LocationResponseDTO gleich sind, kannst du das auch in eins zusammenfassen zu LocationRequestResponseDTO oder einfach LocationDTO